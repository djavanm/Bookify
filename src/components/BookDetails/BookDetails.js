import React from 'react';
import Nav from '../../containers/Nav/Nav';


const BookDetails = ({ currentUser, author_name, book_name, artwork_url, description, release_date, book_id, primary_genre_name }) => {
  const descriptionNew = description.split('<b>').join().split('<i>').join().split('</b>').join().split('</i>').join().split('<br />').join().split(',').join().split(',,,').join()
  const releaseSplit = release_date.split('T')[0].split('-');
  const release = `${releaseSplit[1]}/${releaseSplit[2]}/${releaseSplit[0]}`;
  return (
    <article>
    <Nav currentUser={currentUser} />
    <h2>{book_name}</h2>
    <p>Genre: {primary_genre_name}</p>
    <p>Release: {release}</p>
    <p>About: {descriptionNew}</p>
    </article>
  )
};

export default BookDetails;
