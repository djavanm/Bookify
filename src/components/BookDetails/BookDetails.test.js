import BookDetails from './BookDetails';
import React from 'react';
import { shallow } from 'enzyme';

describe('BookDetails', () => {
  it('should match the snapshot with all of the data passed through', () => {
    const detailsMock = {
      currentUser: '',
      book_name: '',
      description: '',
      release_date: '2019/09/08T24234',
      primary_genre_name: ''
    }
    const wrapper = shallow(<BookDetails {...detailsMock}/>)

    expect(wrapper).toMatchSnapshot();
  });
});