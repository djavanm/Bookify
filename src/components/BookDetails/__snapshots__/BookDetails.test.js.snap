// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetails should match the snapshot with all of the data passed through 1`] = `
<article>
  <Connect(Nav)
    currentUser=""
  />
  <div
    className="details-container"
  >
    <Link
      className="back-btn"
      to="/"
    >
      â—€ back
    </Link>
    <h2 />
    <p>
      Genre: 
    </p>
    <p>
      Release Date: 
      undefined/undefined/2019/09/08
    </p>
    <br />
    <Markup
      content=""
      emptyContent={null}
      parsedContent={null}
      tagName="div"
    />
  </div>
</article>
`;
